cpp/include/cuspatial/detail/join/quadtree_point_in_polygon.cuh

```diff
@@ -118,16 +118,22 @@ std::pair<rmm::device_uvector<IndexType>, rmm::device_uvector<IndexType>> quadtr
 
   auto num_poly_quad_pairs = std::distance(poly_indices_first, poly_indices_last);
 
-  auto quad_lengths_iter =
-    thrust::make_permutation_iterator(quadtree.length_begin(), quad_indices_first);
+  auto quad_lengths_iter = thrust::make_transform_iterator(
+    thrust::make_permutation_iterator(quadtree.length_begin(), quad_indices_first),
+    cuda::proclaim_return_type<std::uint64_t>([] __device__(IndexType const& i) -> std::uint64_t {
+      return static_cast<std::uint64_t>(i);
+    }));
```

Comment:
The quadtree length is a uint32 column, but we have to transform them into uint64 values so the `thrust::inclusive_scan` (below) accumulates into uint64s. Changing the output iterator type to uint64 wasn't sufficient to achieve this behavior.

Reply:
Should this comment be in the code?